### ISSUE TITLE
Make clean doesn't remove all object files and that interferes with profile guided optimization

### ISSUE THREAD

**Issue by i30817** [libretro/beetle-psx-libretro#652]
I was puzzled for a day because profile guided optimization failed compiling with a bunch of ld errors on the second (optimized) build, that was never supposed to link to gcov. If you added the -lgcov manually, it naturally wasn't 'optimized' because it started generating program optimization info to files. If you removed Wl,--no-undefined, it crashed at runtime.

What was it?

make clean on the project root between the two compiles left .o files for some unknown reason, probably the OBJECTS array not having all of the derived built .o files and the 2 builds considering those 'didn't change'. The clean target advertizes it removes all the .o and .d files but it depends on hardcoded lists, lists that are probably not up-to-date.
```

clean:
	@rm -f $(OBJECTS)
	@echo rm -f *.o
	@rm -f $(DEPS)
	@echo rm -f *.d
	rm -f $(TARGET) $(TARGET_TMP)
```

**Comment by i30817** [comment]
BTW, if you're curious (you're probably not), this doesn't help fps much in the hardware renderer, which has abyssal fps because i'm purposefully underclocking the cpu and gpu (overheating). But it does help the software renderer a bit. Increasing the performance of the gpu to max (temporarily just to see) helps it about 10fps, it's a bit underwhelming. Maybe it's more limited by cpu than gpu. 

pcsx-rearmed has sustained 60 fps in hardware or software, though it does say it fallbacks to software if it can't open a hardware contex, so who knows.

**Comment by hizzlekizzle** [comment]
Yeah, I would expect any gains in the software renderer to be mirrored in the hardware renderer since it needs to run the software renderer all the time anyway.

**Comment by i30817** [comment]
I'm curious, if you know, there is any point in repeating the profiling build more than once? Ie: diving this into two steps, one that takes up any input from previous builds (starting with nothing) and builds a exec with profile sampling; so it can be run and built multiple times, collecting info everytime; then the nomal optimized build?

What i'm currently doing is to leave the first build to run in the background in the many games that have a engine 'demo' which is not a movie, which is surprisingly many because Sony wanted the games to run on games unattended on points of sale so they probably created a guideline to put these on start menus, is there any point at all in compiling multiple times collecting info about pressure after previous optimizations or is it pointless?

**Comment by hizzlekizzle** [comment]
I used to do PGO builds for bsnes standalone way back when and my understanding back then was that any changes to the codepath invalidates the previous profiling (and can make it worse than no profiling at all), so I deleted the old profile info and re-ran the tests each time. Dunno if this was actually true, though.
